src/tests/BuildWasmApps/Wasm.Build.Tests/BuildPublishTests.cs

```diff
@@ -0,0 +1,158 @@
+        [BuildAndRun(host: RunHost.V8, aot: true, config: "Release")]
+        [BuildAndRun(host: RunHost.V8, aot: true, config: "Debug")]
+        public void BuildThenPublishWithAOT(BuildArgs buildArgs, RunHost host, string id)
+        {
+            string projectName = $"build_publish_{buildArgs.Config}";
+
+            buildArgs = buildArgs with { ProjectName = projectName };
+            buildArgs = ExpandBuildArgs(buildArgs, extraProperties: "<_WasmDevel>true</_WasmDevel>");
+
+            // no relinking for build
+            bool relinked = false;
+            (_, string output) = BuildProject(buildArgs,
+                                    initProject: () => File.WriteAllText(Path.Combine(_projectDir!, "Program.cs"), s_mainReturns42),
+                                    dotnetWasmFromRuntimePack: !relinked,
+                                    id: id,
+                                    createProject: true,
+                                    publish: false,
+                                    label: "first_build");
+
+            BuildPaths paths = GetBuildPaths(buildArgs);
+            var pathsDict = GetFilesTable(buildArgs, paths, unchanged: false);
+
+            string mainDll = $"{buildArgs.ProjectName}.dll";
+            var firstBuildStat = StatFiles(pathsDict.Select(kvp => kvp.Value.fullPath));
+            Assert.False(firstBuildStat["pinvoke.o"].Exists);
+            Assert.False(firstBuildStat[$"{mainDll}.bc"].Exists);
+
+            CheckOutputForNativeBuild(expectAOT: false, expectRelinking: relinked, buildArgs, output);
+
+            Run(expectAOT: false);
+
+            if (!_buildContext.TryGetBuildFor(buildArgs, out BuildProduct? product))
+                throw new XunitException($"Test bug: could not get the build product in the cache");
+
+            File.Move(product!.LogFile, Path.ChangeExtension(product.LogFile!, ".first.binlog"));
+
+            _testOutput.WriteLine($"{Environment.NewLine}Publishing with no changes ..{Environment.NewLine}");
+            Console.WriteLine($"{Environment.NewLine}Publishing with no changes ..{Environment.NewLine}");
+
+            // relink by default for Release+publish
+            (_, output) = BuildProject(buildArgs,
+                                    id: id,
+                                    dotnetWasmFromRuntimePack: false,
+                                    createProject: false,
+                                    publish: true,
+                                    useCache: false,
+                                    label: "first_publish");
+
+            var publishStat = StatFiles(pathsDict.Select(kvp => kvp.Value.fullPath));
+            Assert.True(publishStat["pinvoke.o"].Exists);
+            Assert.True(publishStat[$"{mainDll}.bc"].Exists);
+            CheckOutputForNativeBuild(expectAOT: true, expectRelinking: false, buildArgs, output);
+            CompareStat(firstBuildStat, publishStat, pathsDict.Values);
+
+            Run(expectAOT: true);
```

Comment:
Between this and line 99, it seems like the tests get run multiple times in the same run with the same name.  Needs disambiguating or it breaks ingestion to Kusto.

Reply:
Um it's one test though, so I would expect one entry for the test run. And it will have total of two runs, one per config (Debug, and Release). Or did I misunderstand you?

The `Run` method here is running the built wasm app.

Reply:
don't mind me, this investigation led to finding another issue :)